code/datums/components/rngod.dm

```diff
@@ -0,0 +1,95 @@
+  *
+  *	Currently blessing and jinxing only affects your rolls for hitting a vending machine and getting loot/killed, but if there's interest, rolls with degrees of success can be added
+  *		in other places as well. It's something I'd like to do!
+  *
+  * Arguments: none. why, are we really gonna fight over this right now in front of our guests?
+  * *
+  */
+/datum/component/rngod
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	/// when did we last bless someone?
+	var/lastBless
+	/// when did we last curse someone?
+	var/lastCurse
+
+	/// how long do we have to wait between bless attempts?
+	var/blessCooldown = 60 SECONDS
+	/// how long do we have to wait between curse attempts?
+	var/curseCooldown = 45 SECONDS
+
+
+/datum/component/rngod/Initialize()
+
+/datum/component/rngod/Destroy(force, silent)
+	return ..()
+
+/datum/component/rngod/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOB_SAY, .proc/checkSpeech)
+
+/datum/component/rngod/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOB_SAY)
+
+/datum/component/rngod/proc/checkSpeech(mob/O, list/speech_args)
+	var/message = lowertext(speech_args[SPEECH_MESSAGE])
+	var/mob/target
+
+	if(findtext(message, "damn") && world.time > lastCurse + curseCooldown)
+		lastCurse = world.time
+		target = findTarget(message)
+		if(!target)
+			target = pick(viewers(parent)) || parent // if we can't curse anyone else, at least we can curse ourself!
+
+		if(target.mind && target.mind.fate)
+			target.mind.fate.curse(parent, 10)
+
+	else if(findtext(message, "bless") && world.time > lastBless + blessCooldown)
+		lastBless = world.time // even if we don't get someone, still charge the card
+		target = findTarget(message)
+		if(!target)
+			target = pick(oviewers(parent)) // can't bless yourself like you can damn yourself!
+
+		if(target == parent) // in case we tried explicitly naming ourself
+			return
+
+		if(target && target.mind && target.mind.fate) // check if there's a target again in case we couldn't find someone in our random nearby person check
+			target.mind.fate.bless(parent, 10)
+
+/**
+  * Find a mob in view and a specified radius based on the words in our message. If we name someone specifically along with our bless/curse word, they'll be who we return.
+  *
+  *	Code lovingly stolen and modified for the worse from adminhelp.dm's keywords_lookup proc
+  *
+  *
+  * Arguments:
+  * *msg- The lowercase'd message string we're digging through for names
+  * *radius- How far we're looking
+  */
+/datum/component/rngod/proc/findTarget(msg, radius=3)
+	//explode the input msg into a list
+	var/list/msglist = splittext(msg, " ")
+
+	for(var/mob/M in view(radius, parent)) // look through all the mobs nearby
+		var/list/nameWords = list()
+		if(!M.mind || !M.mind.fate)
+			continue
+
+		for(var/string in splittext(lowertext(M.real_name), " "))
+			nameWords += string
+		for(var/string in splittext(lowertext(M.name), " "))
+			nameWords += string
```

Comment:
i made a first_name() helper for voice of god, you could check it here so you can curse someone without having to get their full name right

Reply:
`M.first_name()`

Reply:
It splits on spaces anyway, so getting any of the individual parts of their name works

Reply:
ah, i didn't even notice the splittext, my bad

it doesn't catch lizard names, but you could fix that by replacing dashes with spaces beforehand

Reply:
some lizards don't go by their first names though (c.f. Is-a-Lizard)

Reply:
what happens when people share individual parts of names, like, Eats-the-Lemons and Smells-the-Feet

Reply:
These are the RNGods, they just stochastically pick one! Well, whoever comes up first in the search anyway

Reply:
is viewers pre-randomized? (can people powergame their names to avoid this)

Reply:
imagine powergaming your name so the chaplain can't make it slightly more likely that a vending machine squishes you

plus, if you make it more likely to get cursed you also make it less likely to get blessed and viceversa

Reply:
You should still shuffle it if it's just based on alphabetical order, though.